roofit/roofitcore/src/RooAbsCollection.cxx

```diff
@@ -318,12 +318,27 @@ Bool_t RooAbsCollection::addServerClonesToList(const RooAbsArg& var)
 
 
 ////////////////////////////////////////////////////////////////////////////////
-/// The assignment operator sets the value of any argument in our set
-/// that also appears in the other set.
+/// The copy assignment operator redirects to assign(). This is confusing
+/// because it's not a properly implemented copy assignment operator, also
+/// because it's inconsistent with the copy constructor. Unfortunately, it is too
+/// late to change this behaviour now. If possible, please use assign()
+/// explicitly in your code to make the intention clear that you don't want to
+/// make a copy-assignment.
```

Comment:
```
/// Assign values from the elements in `other` to our elements.
/// \warning This is not a conventional assignment operator. To avoid confusion, prefer using RooAbsCollection::assign().
```